pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -209,18 +221,16 @@ func (os *OpenStack) DetachVolume(instanceID, volumeID string) error {
 		return fmt.Errorf("can not detach volume %s, its status is %s", volume.Name, volume.Status)
 	}
 
+	// Incase volume is of type multiattach, it could be attached to more than one instance
 	if len(volume.Attachments) > 0 {
-		if volume.Attachments[0].ServerID != instanceID {
-			return fmt.Errorf("disk: %s is not attached to compute: %s", volume.Name, instanceID)
-		}
 		err = volumeattach.Delete(os.compute, instanceID, volume.ID).ExtractErr()
```

Comment:
We still need to check if the volume has been attached to the instance.

Reply:
ya, I suppose it could be done, loop through attachments.  I felt it should be handled by delete call anyway. I can add the check